src/server/internalServer.cpp

```diff
@@ -138,11 +138,12 @@ InternalServer::InternalServer(Library* library,
   m_withTaskbar(withTaskbar),
   m_withLibraryButton(withLibraryButton),
   m_blockExternalLinks(blockExternalLinks),
-  m_indexTemplateString(indexTemplateString),
   mp_daemon(nullptr),
   mp_library(library),
   mp_nameMapper(nameMapper ? nameMapper : &defaultNameMapper)
-{}
+{
+  m_indexTemplateString = !indexTemplateString.empty() ? indexTemplateString : RESOURCE::templates::index_html;
```

Comment:
It is better to have a "positive" test.
We can always set `m_indexTemplateString` to `indexTemplateString` in the initializer list and then, in the constructor body, test if `m_indexTemplateString` is empty and then set it to `RESOURCE::templates::index_html`